src/main/java/io/anserini/util/MapCollections.java

```diff
@@ -98,50 +101,22 @@ private MapThread(DocumentCollection collection, Path inputFile) {
     @Override
     public void run() {
       try {
-        int numMapped = 0;
-
         @SuppressWarnings("unchecked")
         BaseFileSegment<SourceDocument> iter =
           (BaseFileSegment) ((SegmentProvider) collection).createFileSegment(inputFile);
 
-        while (true) {
-          boolean hasNext;
-          try {
-            hasNext = iter.hasNext();
-          } catch (NoSuchElementException e1) {
-            break;
-          } catch (RuntimeException e2) {
-            if (e2.getMessage() != null && e2.getMessage().contains("IOException")) {
-              LOG.error("Exception when parsing document: ", e2);
-              if (mapper.isCountDocumentMapper()) {
-                ((CountDocumentMapper)mapper).incrementErrors();
-              }
-              break; // IOException: stop reading more documents
-            } else {
-              if (mapper.isCountDocumentMapper()) {
-                ((CountDocumentMapper)mapper).incrementSkipped();
-              }
-              continue; // Non-IOException: continue reading the next document
-            }
-          }
-
-          if (!hasNext) {
-            break;
-          }
-
-          SourceDocument d = iter.next();
-
-          boolean isProcessed = mapper.process(d);
-          if (!isProcessed) {
-            continue;
-          }
-
-          numMapped++;
-        }
+        // We're calling these records because the documents may not in indexable.
+        AtomicInteger records = new AtomicInteger();
+        try {
+          iter.forEachRemaining(d -> {
+            mapper.process(d, context);
+            records.incrementAndGet();
+          });
+        } catch (NoSuchElementException e) {}
```

Comment:
Se we are not catching `RuntimeException` here?

Reply:
okay, will do.

Reply:
Actually, I just noticed, there's another catch block below:
```
catch (Exception e) {
        LOG.error(Thread.currentThread().getName() + ": Unexpected Exception:", e);
      }
```
This will catch all exceptions for this file segment, so I think we're good.

Reply:
I think we can remove this `try...catch...` block after #399 is merged.